Sources/SQLiteNIO/SQLiteCustomFunction.swift

```diff
@@ -0,0 +1,404 @@
+                return FunctionDefinition(compute: compute)
+            case .aggregate(let makeAggregate):
+                return AggregateDefinition(makeAggregate: makeAggregate)
+            }
+        }
+
+        /// Feeds the `xFunc` parameter of sqlite3_create_function_v2
+        /// http://sqlite.org/capi3ref.html#sqlite3_create_function
+        var xFunc: (@convention(c) (OpaquePointer?, Int32, UnsafeMutablePointer<OpaquePointer?>?) -> Void)? {
+            guard case .function = self else { return nil }
+            return { (sqliteContext, argc, argv) in
+                let definition = Unmanaged<FunctionDefinition>
+                    .fromOpaque(sqlite_nio_sqlite3_user_data(sqliteContext))
+                    .takeUnretainedValue()
+                do {
+                    try SQLiteCustomFunction.report(
+                        result: definition.compute(argc, argv),
+                        in: sqliteContext)
+                } catch {
+                    SQLiteCustomFunction.report(error: error, in: sqliteContext)
+                }
+            }
+        }
+
+        /// Feeds the `xStep` parameter of sqlite3_create_function_v2
+        /// http://sqlite.org/capi3ref.html#sqlite3_create_function
+        var xStep: (@convention(c) (OpaquePointer?, Int32, UnsafeMutablePointer<OpaquePointer?>?) -> Void)? {
+            guard case .aggregate = self else { return nil }
+            return { (sqliteContext, argc, argv) in
+                let aggregateContextU = SQLiteCustomFunction.unmanagedAggregateContext(sqliteContext)
+                let aggregateContext = aggregateContextU.takeUnretainedValue()
+                assert(!aggregateContext.hasErrored) // assert SQLite behavior
+                do {
+                    let count = Int(argc)
+                    let arguments = try (0 ..< count).map { index -> SQLiteData in
+                        guard let value = argv?[index] else {
+                            throw SQLiteCustomFunctionArgumentError(count: count, index: index)
+                        }
+                        return try SQLiteData(sqliteValue: value)
+                    }
+                    try aggregateContext.aggregate.step(arguments)
+                } catch {
+                    aggregateContext.hasErrored = true
+                    SQLiteCustomFunction.report(error: error, in: sqliteContext)
+                }
+            }
+        }
+
+        /// Feeds the `xFinal` parameter of sqlite3_create_function_v2
+        /// http://sqlite.org/capi3ref.html#sqlite3_create_function
+        var xFinal: (@convention(c) (OpaquePointer?) -> Void)? {
+            guard case .aggregate = self else { return nil }
+            return { (sqliteContext) in
+                let aggregateContextU = SQLiteCustomFunction.unmanagedAggregateContext(sqliteContext)
+                let aggregateContext = aggregateContextU.takeUnretainedValue()
+                aggregateContextU.release()
```

Comment:
This seems like a roundabout way of spelling `takeRetainedValue`.

Reply:
I couldn't agree more 😅